docs/getting-started-guides/mesos.md

```diff
@@ -0,0 +1,297 @@
+
+Running Kubernetes on Mesos allows you to easily move Kubernetes workloads from one cloud provider to another to your own physical datacenter.
+
+This tutorial will walk you through setting up Kubernetes on a Mesos cluster on [Google Cloud Plaform][4].
+It provides a step by step walk through of adding Kubernetes to a Mesos cluster and running the classic GuestBook demo application.
+The walkthrough presented here is based on the v0.4.x series of the Kubernetes-Mesos project, which itself is based on Kubernetes v0.11.0.
+
+### Prerequisites
+
+* Understanding of [Apache Mesos][11]
+* Mesos cluster on [Google Compute Engine][5]
+* Identify the Mesos master node external IP from Mesosphere [cluster launch pad][12]
+* A [VPN connection to the cluster][6].
+
+### Deploy Kubernetes-Mesos
+
+Log into the master node over SSH, replacing the placeholder below with the correct IP address.
+
+```bash
+ssh jclouds@${ip_address_of_master_node}
+```
+
+Build Kubernetes-Mesos.
+
+```bash
+$ git clone https://github.com/mesosphere/kubernetes-mesos k8sm
+$ mkdir -p bin && sudo docker run --rm -v $(pwd)/bin:/target \
+  -v $(pwd)/k8sm:/snapshot -e GIT_BRANCH=release-0.4 \
+  mesosphere/kubernetes-mesos:build
+```
+
+Set some environment variables.
+The internal IP address of the master is visible via the cluster details page on the Mesosphere launchpad, or may be obtained via `hostname -i`.
+
+```bash
+$ export servicehost=$(hostname -i)
+$ export mesos_master=${servicehost}:5050
+$ export KUBERNETES_MASTER=http://${servicehost}:8888
+```
+### Deploy etcd
+Start etcd and verify that it is running:
+
+```bash
+$ sudo docker run -d --hostname $(hostname -f) --name etcd -p 4001:4001 -p 7001:7001 coreos/etcd
+```
+
+```bash
+$ sudo docker ps
+CONTAINER ID   IMAGE                COMMAND   CREATED   STATUS   PORTS                NAMES
+fd7bac9e2301   coreos/etcd:latest   "/etcd"   5s ago    Up 3s    2379/tcp, 2380/...   etcd
+```
+It's also a good idea to ensure your etcd instance is reachable by testing it
+```bash
+curl -L http://$servicehost:4001/v2/keys/
+```
+If connectivity is OK, you will see an output of the available keys in etcd (if any).
+
+### Start Kubernetes-Mesos Services
+Start the kubernetes-mesos API server, controller manager, and scheduler on a Mesos master node:
+
+```bash
+$ ./bin/km apiserver \
+  --address=${servicehost} \
+  --mesos_master=${mesos_master} \
+  --etcd_servers=http://${servicehost}:4001 \
+  --portal_net=10.10.10.0/24 \
+  --port=8888 \
+  --cloud_provider=mesos \
+  --v=1 >apiserver.log 2>&1 &
+
+$ ./bin/km controller-manager \
+  --master=$servicehost:8888 \
+  --mesos_master=${mesos_master} \
+  --v=1 >controller.log 2>&1 &
+
+$ ./bin/km scheduler \
+  --address=${servicehost} \
+  --mesos_master=${mesos_master} \
+  --etcd_servers=http://${servicehost}:4001 \
+  --mesos_user=root \
+  --api_servers=$servicehost:8888 \
+  --v=2 >scheduler.log 2>&1 &
+```
+
+Start the service proxy on each Mesos slave node:
```

Comment:
You might want to explain briefly that the Mesos executor will start up the kubelet, since users might wonder why you're not also starting up the kubelet.